configuration/exodus-pipeline-resources.yaml

```diff
@@ -219,3 +229,44 @@ Resources:
       Artifacts:
         Type: NO_ARTIFACTS
       TimeoutInMinutes: 10
+
+  WhitelistUploadResource:
+    Type: Custom::WhitelistUploadFunction
+    Condition: UploadWhitelist
+    Properties:
+      ServiceToken: !GetAtt WhitelistUploadFunction.Arn
+      DestBucketName: !Sub ${project}-pipeline-artifacts
+      DestBucketRegion: !Ref region
+      WhitelistUrl: !Ref whitelistUrl
+    DependsOn:
+      - WhitelistUploadFunction
+
+  WhitelistUploadFunction:
+    Type: AWS::Lambda::Function
+    Condition: UploadWhitelist
+    Properties:
+      Handler: index.handler
+      Role: !Ref lambdafunctionrole
+      FunctionName: !Sub ${project}-whitelist-upload
+      Code:
+        ZipFile: |
+          import urllib.request
+          import cfnresponse
+          import boto3
+
+
+          def handler(event, context):
+              bucket = event["ResourceProperties"]["DestBucketName"]
+              region = event["ResourceProperties"]["DestBucketRegion"]
+              whitelist_url = event["ResourceProperties"]["WhitelistUrl"]
+
+              # Upload whitelist only when this resource is created
+              if event["RequestType"] == "Create":          
```

Comment:
I'm not familiar with how the RequestType is applied.  Can this also be used to reset the whitelist on demand? e.g. imagine a commit accidentally breaks formatting of the whitelist file and locks us all out from deployment by normal means.  Can we fix it by rerunning this pointing at a correct whitelist, or would it not be possible to trigger the "Create" condition?

Reply:
"Create" only happens once, when the resource is initially created, but "Update" is another RequestType. If "Update" was added here, the whitelist would be overwritten each time this stack is updated.